src/AutoRest.CSharp/LowLevel/Decorator/MethodAmbiguityResolver.cs

```diff
@@ -0,0 +1,169 @@
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using AutoRest.CSharp.Generation.Types;
+using AutoRest.CSharp.Input.Source;
+using AutoRest.CSharp.Output.Models;
+using AutoRest.CSharp.Output.Models.Shared;
+using AutoRest.CSharp.Output.Models.Types;
+using Microsoft.CodeAnalysis;
+
+namespace AutoRest.CSharp.LowLevel.Decorator
+{
+    internal static class MethodAmbiguityResolver
+    {
+        /// <summary>
+        /// Resolve ambiguous methods between DPG client and existing client codes.
+        /// </summary>
+        /// <param name="clientMethods">Client methods belonging to a DPG client.</param>
+        /// <param name="clientNamespace">Namespace of the DPG client.</param>
+        /// <param name="clientName">Name of the DPG client.</param>
+        /// <param name="sourceInputModel">Roslyn source model for existing client codes.</param>
+        /// <returns>An array of LowLwevelClientMethod which have been resolved to avoid ambiguity with existing client codes.</returns>
+        internal static LowLevelClientMethod[] ResolveClientMethods(LowLevelClientMethod[] clientMethods, string clientNamespace, string clientName, SourceInputModel sourceInputModel)
+        {
+            var clientSourceModel = sourceInputModel.FindForType(clientNamespace, clientName);
+            if (clientSourceModel != null)
+            {
+                var newClientMethods = new LowLevelClientMethod[clientMethods.Length];
+                for (int i = 0; i < clientMethods.Length; i++)
+                {
+                    newClientMethods[i] = ResolveClientMethod(clientMethods[i], clientSourceModel);
+                }
+                return newClientMethods;
+            }
+
+            return clientMethods;
+        }
+
+        /// <summary>
+        /// Reslove ambiguity between a DPG client method and client methods under the client with the same name as DPG client in existing codes.
+        /// </summary>
+        /// <param name="clientMethod">Client method belonging  to a DPG client.</param>
+        /// <param name="clientSourceModel">Roslyn model for a client with the same name of DPG client in existing codes.</param>
+        /// <returns>A LowLevelClientMethod which have been resolved to avoid ambiguity with the methods in the given client in existing codes.</returns>
+        private static LowLevelClientMethod ResolveClientMethod(LowLevelClientMethod clientMethod, INamedTypeSymbol clientSourceModel)
+        {
+            var clientMethodSourceModels = clientSourceModel.GetMembers().Where(s => s.Kind == SymbolKind.Method &&
+                s.Name == clientMethod.Signature.Name && s.DeclaredAccessibility == Accessibility.Public).Select(s => (IMethodSymbol)s).ToList();
+            if (clientMethodSourceModels.Count > 0)
+            {
+                return ResolveClientMethod(clientMethod, clientMethodSourceModels);
+            }
```

Comment:
We actually do not need to convert this to a list (you did not use indices).
if you want to check if this is empty, you could use `clientMethodSourceModels.Any()` which is more efficient.

Also I would suggest we could wrap the condition here to a method to increase the readibility